tests/test_branch_command.rs

```diff
@@ -113,6 +114,63 @@ fn test_branch_forget_glob() {
     "###);
 }
 
+#[test]
+fn test_branch_forget_export() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    test_env.jj_cmd_success(&repo_path, &["new"]);
+    test_env.jj_cmd_success(&repo_path, &["branch", "set", "foo"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["branch", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    foo: 65b6b74e0897 (no description set)
+    "###);
+
+    // Exporting the branch to git creates a local-git tracking branch
+    let stdout = test_env.jj_cmd_success(&repo_path, &["git", "export"]);
+    insta::assert_snapshot!(stdout, @"");
+    let stdout = test_env.jj_cmd_success(&repo_path, &["branch", "forget", "foo"]);
+    insta::assert_snapshot!(stdout, @"");
+    // Forgetting a branch does not delete its local-git tracking branch. This is
+    // the opposite of what happens to remote-tracking branches.
+    // TODO: Consider allowing forgetting local-git tracking branches as an option
+    let stdout = test_env.jj_cmd_success(&repo_path, &["branch", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    foo (deleted)
+      @git: 65b6b74e0897 (no description set)
+    "###);
+
+    // Aside: the way we currently resolve git refs means that `foo`
+    // resolves to `foo@git` when actual `foo` doesn't exist.
+    // TODO: Decide this is a bug and fix it?
```

Comment:
I know that functionality was intentional but that was implemented a long time ago, maybe even before `jj git import` existed. I think it makes sense to drop that functionality now. I think it's useful to be able to still check out git tags (e.g. `jj co v.1.0.0`). Maybe we should make the ref-importing code create tags.

Reply:
I'll try to add a commit disabling it for git-tracking of local branches.

Reply:
And done.